UEFI working prototype

ainfosec-henselb opened the pull request.

rianquinn commented on src/arch/uefi/FileAccess.c:
```diff
@@ -0,0 +1,258 @@
+
+
+
+// LoadFile/2 protocol doesn't seem to work for arbitrary file
```
remove unused code

rianquinn commented on src/arch/uefi/HypervisorPkg.dsc:
```diff
@@ -0,0 +1,83 @@
+## @file
+#   Intel(r) UEFI Standard Libraries for EDK II.
+#   Build descriptions.
```
If these files are needed keep them, if they are generated please remove them.

rianquinn commented on src/arch/uefi/Platform.c:
```diff
@@ -0,0 +1,282 @@
+ */
+void* platform_virt_to_phys(void* virt)
+{
+  return virt;
```
lol... that's awesome

rianquinn commented on src/arch/uefi/Platform.c:
```diff
@@ -0,0 +1,282 @@
+ */
+void platform_start(void)
+{
+  ;
```
no need for ;

rianquinn commented on src/arch/uefi/Platform.c:
```diff
@@ -0,0 +1,282 @@
+ */
+void platform_stop(void)
+{
+  ;
```
same as above

rianquinn commented on src/arch/uefi/Platform.c:
```diff
@@ -0,0 +1,282 @@
+  if (!EFI_ERROR(status)) {
+    return N;
+  } else {
+    return 0;
```
in general, Clang Tidy will complain about your `else` statement here as it is not needed.

rianquinn commented on src/arch/uefi/Platform.c:
```diff
@@ -0,0 +1,282 @@
+
+void platform_restore_preemption(void)
+{
+  ;
```
not needed

rianquinn commented on src/arch/uefi/PrepareProcessor.c:
```diff
@@ -0,0 +1,95 @@
+
+  _sgdt((void*)&Gdtr.Limit);
+
+  void* Gdt = (void*)Gdtr.Base;
```
need to run astyle against your code. If it is not working let me know so that we can fix that. Our format style is `void *`, not `void*`

rianquinn commented on src/arch/uefi/vmcall.nasm:
```diff
@@ -0,0 +1,94 @@
+    ret
+
+
+;platform_vmcall_event PROC
```
remove

ainfosec-henselb commented:
Think I got it fixed up.  

Switched my formatter to 4 spaces and name-side asterisks.  
The .dsc, .dec, and .inf files are not auto-generated and are needed for the EDK build.  
Did a bunch of whitespace and dead/commented code removal and misc cleanup (didn't realize it was as bad as it was - oops)
Added licenses
Removed function documentation from platform.c as it was copied in from platform.h
